File: inventario-app/src/app/modelos/producto.ts
export interface Categoria {
  id: number;
  nombre: string;
  descripcion?: string;
  colorIdentificador?: string;
  icono?: string;
}

export enum UnidadMedida {
  UNIDAD = 'UNIDAD',
  CAJA = 'CAJA',
  PAQUETE = 'PAQUETE',
  RESMA = 'RESMA',
  LITRO = 'LITRO',
  KILO = 'KILO',
  METRO = 'METRO'
}

export interface Producto {
  id: number;
  codigo: string;
  nombre: string;
  descripcion?: string;
  categoria: Categoria;
  unidadMedida: UnidadMedida;
  precioUnitario: number;
  stockMinimo: number;
  stockMaximo: number;
  imagen?: string;
  estado: string;
}

File: src/main/java/com/utp/integrador/clinica/model/Producto.java
package com.utp.integrador.clinica.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import lombok.Data;

@Data
@Entity
@Table(name = "productos")
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "producto_id")
    private Integer id;

    @Column(length = 50, nullable = false, unique = true)
    private String codigo;

    @Column(length = 200, nullable = false)
    private String nombre;

    @Column(columnDefinition = "TEXT")
    private String descripcion;

    @ManyToOne
    @JoinColumn(name = "categoria_id", nullable = false)
    private Categoria categoria;

    @Enumerated(EnumType.STRING)
    @Column(name = "unidad_medida", nullable = false)
    private UnidadMedida unidadMedida;

    @Column(name = "precio_unitario", precision = 10, scale = 2, nullable = false)
    private BigDecimal precioUnitario;

    @Column(name = "stock_minimo", nullable = false)
    private Integer stockMinimo;

    @Column(name = "stock_maximo", nullable = false)
    private Integer stockMaximo;

    @Column(length = 500)
    private String imagen;

    @Column(length = 10, nullable = false)
    private String estado = "Activo";
}
